assets/src/blocks/components/utilities/index.js

```diff
@@ -0,0 +1,126 @@
+/* global wcSettings */
+/**
+ * Fast button block utilities.
+ */
+
+import { flatten, uniqBy } from 'lodash';
+
+const apiFetch = wp.apiFetch;
+const { addQueryArgs } = wp.url;
+
+const productsBase = '/wc/store/products';
+
+/**
+ * Get the number of products in the WooCommerce store from the global wcSettings object.
+ */
+const getProductCount = () => {
+	const wcSettings = window.wcSettings || {};
+	const blocksConfig = wcSettings.hasOwnProperty( 'wcBlocksConfig' ) ? wcSettings.wcBlocksConfig : {};
+	const productCount = blocksConfig.hasOwnProperty( 'productCount' ) ? blocksConfig.productCount : 100;
+
+	return productCount;
+};
+
+/**
+ * Get a promise that resolves to a list of products from the WooCommerce Store API.
+ *
+ * @param {Object} request          A query object with the list of selected products and search term.
+ * @param {Array}  request.selected Currently selected products.
+ * @param {string} request.search   Search string.
+ */
+export const getProducts = (
+	{
+		selected = 0,
+		search = '',
+	}
+) => {
+	const productCount = getProductCount();
+	const isLargeCatalog = productCount > 100;
+
+	const queryArgs = {
+		per_page: isLargeCatalog ? 100 : 0,
+		catalog_visibility: 'any',
+		search,
+		orderby: 'title',
+		order: 'asc',
+	};
+
+	const requests = []
+
+	// Add a query to make sure all selected products are included in the response.
+	if ( selected ) {
+		const include = [ selected ];
+		requests.push(
+			addQueryArgs(
+				productsBase,
+				{
+					catalog_visibility: 'any',
+					include,
+				}
+			)
+		);
+	}
+
+	requests.push( addQueryArgs( productsBase, queryArgs ) );
+
+	return Promise.all( requests.map( ( path ) => apiFetch( { path } ) ) )
+		.then( ( data ) => {
+			const products = uniqBy( flatten( data ), 'id' );
```

Comment:
should just be able to use the built-in `flat` function here
```suggestion
			const products = uniqBy( data.flat(), 'id' );
```